misc/toml-parser/src/main/java/io/ballerina/toml/validator/schema/Schema.java

```diff
@@ -58,9 +65,7 @@ public static Schema from(Path jsonPath) throws IOException {
         Gson gson = new GsonBuilder()
                 .registerTypeAdapter(AbstractSchema.class, new SchemaDeserializer()).create();
         BufferedReader reader = Files.newBufferedReader(jsonPath);
-        Schema rootSchema = gson.fromJson(reader, Schema.class);
-        rootSchema.setType(Type.OBJECT);
-        return rootSchema;
+        return (Schema) gson.fromJson(reader, AbstractSchema.class);
```

Comment:
Seems like this casting is unnecessary

Reply:
Casting is needed as the static method should return a ObjectSchema(Schema) instead of AbstractSchema. 
Note in the old code we called `gson.fromJson(reader, Schema.class);` and now we call `gson.fromJson(reader, AbstractSchema.class);`